sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -1380,5 +1380,314 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
                 throw;
             }
         }
+
+        /// <summary>
+        /// Creates Time Series instances types asynchronously. If a provided instance type is already in use, then this will attempt to replace the existing instance type with the provided Time Series Instance.
+        /// </summary>
+        /// <param name="timeSeriesTypes">The Time Series instances types to be created or replaced.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of error objects corresponding by position to the <paramref name="timeSeriesTypes"/> array in the request.
+        /// An error object will be set when operation is unsuccessful.
+        /// </returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesTypes"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesTypes"/> is empty.
+        /// </exception>
+        public virtual async Task<Response<TimeSeriesOperationError[]>> CreateOrReplaceTimeSeriesTypesAsync(
```

Comment:
Can it really be "OrReplace" if there is no way to pass an ETag?

Reply:
From what I recall, it does still replace. Without the etag, there is just no concurrency protection around it

Reply:
yes if anything changes on the type parameter, it does replace.